### DISCUSSION TITLE
Main Settings

### DISCUSSION THREAD

**Discussion by infamous-panda** [terjeio/ioSender#158] [General]
I don't see it in the Wiki but is there link that explains the function of the Main settings and options

![image](https://user-images.githubusercontent.com/54411216/142249856-0d92eed8-9340-460d-adbc-e196e85df410.png)

Some are more obvious than others but for instance why would someone want to change the Max buffer or Poll interval. Or Auto-compress gcode?

**Comment by terjeio** [comment]
There is a lot of documentation to write...

If you hover over the settings there is a little bit of extra info.

_Max buffer_ is how much of the controller input buffer is used by agressive buffering before waiting for space.
I sometimes want to adjust this setting when testing, could be useful for others if on a slow connection.
Most grblHAL drivers has a 1K input buffer. Vanilla Grbl has just 128 bytes.

_Poll interval_ is how often the controller is queried for the status report, so this is how often the UI is updated with positions etc. AFAIK this is a common setting across senders.

_Auto-compress_ will often reduce gcode size, sometimes significantly. This setting is for applying auto compress when doing gcode transfors such as adding drag knife moves. Compression can also be applied via _File > Transform > Compres_.
Compression just removes gcode commands/parameters that is repeated, e.g this uncommpressed fragment:
```
G00X0.0000Y0.0000F2400.0
G00X3.2999Y2.9559
G1X3.2999Y2.9559F2499.8
G1X4.6573Y2.9559F2400.0
G1X5.0472Y2.9559
G1X4.8322Y3.0267
G1X4.2824Y3.1862
G1X3.9408Y3.2545
G1X3.4568Y3.2795
G1X3.2999Y3.1727
G1X3.2999Y2.9559
G00X3.2999Y2.9559
G00X3.2999Y4.9716
```
will be "compressed" to:
```
F2400G0X0Y0
X3.2999Y2.9559
F2499.8G1X3.2999Y2.9559
F2400X4.6573
X5.0472
X4.8322Y3.0267
X4.2824Y3.1862
X3.9408Y3.2545
X3.4568Y3.2795
X3.2999Y3.1727
Y2.9559
G0X3.2999Y2.9559
Y4.9716
```
